example-application/test/basic-tests.js

```diff
@@ -95,6 +95,7 @@ describe("/api", () => {
         id: 1,
         name: "John",
       });
+      nock("http://localhost/").post(`/mailer`).reply(202);
```

Comment:
Totally.  I think this differentiation should be a best-practice as well!
I think a stub is a correct approach here. I'll fix it.

Reply:
@goldbergyoni Can you elaborate more your idea? as teams usually use `nock` is for intercepting 3rd parties calls. The way we have used among projects is recording this calls and use it as fixtures in the next execution.

Reply:
@Thormod

Differentiate between two scenarios:
1. A:Our code makes HTTP call - Our Order Microservice is calling the User Microservice, we are aware of this call and it is unlikely to change without our notice
2. B: Other code is making HTTP call - We use some npm lib, like AWS SES, that under the good is making some HTTP call. This might change without our notice

To my understanding, A is valid for knocking, B not